eth/backend.go

```diff
@@ -374,27 +375,54 @@ func (s *Ethereum) ResetWithGenesisBlock(gb *types.Block) {
 	s.blockchain.ResetWithGenesisBlock(gb)
 }
 
-func (s *Ethereum) Etherbase() (eb common.Address, err error) {
+func (s *Ethereum) Validator() (val common.Address, err error) {
+	s.lock.RLock()
+	validator := s.validator
+	s.lock.RUnlock()
+
+	if validator != (common.Address{}) {
+		return validator, nil
+	}
+	if wallets := s.AccountManager().Wallets(); len(wallets) > 0 {
+		if accounts := wallets[0].Accounts(); len(accounts) > 0 {
+			validator := accounts[0].Address
+
+			s.lock.Lock()
+			s.validator = validator
+			s.lock.Unlock()
+
+			log.Info("Validator automatically configured", "address", validator)
+			return validator, nil
+		}
+	}
+	return common.Address{}, fmt.Errorf("validator must be explicitly specified")
+}
+
+func (s *Ethereum) TxFeeRecipient() (common.Address, error) {
 	s.lock.RLock()
-	etherbase := s.etherbase
+	txFeeRecipient := s.txFeeRecipient
 	s.lock.RUnlock()
 
-	if etherbase != (common.Address{}) {
-		return etherbase, nil
+	if txFeeRecipient != (common.Address{}) {
+		return txFeeRecipient, nil
 	}
 	if wallets := s.AccountManager().Wallets(); len(wallets) > 0 {
 		if accounts := wallets[0].Accounts(); len(accounts) > 0 {
-			etherbase := accounts[0].Address
+			txFeeRecipient := accounts[0].Address
 
 			s.lock.Lock()
-			s.etherbase = etherbase
+			s.txFeeRecipient = txFeeRecipient
 			s.lock.Unlock()
 
-			log.Info("Etherbase automatically configured", "address", etherbase)
-			return etherbase, nil
+			log.Info("TxFeeRecipient automatically configured", "address", txFeeRecipient)
+			return txFeeRecipient, nil
 		}
 	}
-	return common.Address{}, fmt.Errorf("etherbase must be explicitly specified")
+	return common.Address{}, fmt.Errorf("txFeeRecipient must be explicitly specified")
+}
+
+func (s *Ethereum) Etherbase() (common.Address, error) {
+	return s.Validator()
```

Comment:
In `eth/api.go` the `Etherebase` function routes to the tx fee recipient. This is inconsistent with that, and it seems like changing this to tx fee recipient would have better sematics.

Reply:
It appears `ethstats` uses this address to sign data, do you know what address would be idiomatic to use here?
I would assume `validator`, but agree this isn't great semantics.
If you agree that the signer should be `validator` maybe I'll change this from `etherbase` to `validator()`

Reply:
It uses the validator signer address, so making this `Validator` would work well